backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Docs_Generics_BPartner_Report.sql

```diff
@@ -0,0 +1,165 @@
+/*
+ * #%L
+ * de.metas.fresh.base
+ * %%
+ * Copyright (C) 2023 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+DROP FUNCTION IF EXISTS de_metas_endcustomer_fresh_reports.docs_generics_bpartner_report(p_org_id    numeric,
+                                                                                         p_doctype   text,
+                                                                                         p_bp_loc_id numeric,
+                                                                                         p_record_id numeric)
+;
+
+CREATE FUNCTION de_metas_endcustomer_fresh_reports.docs_generics_bpartner_report(p_org_id    numeric,
+                                                                                 p_doctype   text,
+                                                                                 p_bp_loc_id numeric,
+                                                                                 p_record_id numeric)
+
+    RETURNS TABLE
+            (
+                org_name        text,
+                org_addressline text,
+                address1        text,
+                postal          text,
+                city            text,
+                country         text,
+                gln             text,
+                addressblock    text
+            )
+    STABLE
+    LANGUAGE sql
+
+AS
+$$
+SELECT x.org_name,
+       x.org_addressline,
+       x.address1,
+       x.postal,
+       x.city,
+       x.country,
+       bpl.gln,
+
+       CASE
+           WHEN p_bp_loc_id IS NOT NULL
+               THEN
+                   COALESCE(bp.name || E'\n', '') || COALESCE(bpl.address, '')
+           WHEN p_doctype = 'shn'
+               THEN
+                   COALESCE(bp.name || E'\n', '') || COALESCE(bpl.address, '')
+           WHEN p_doctype = 'o'
```

Comment:
pls, take out bp.name
bpl.address shall already contain the name

Reply:
done